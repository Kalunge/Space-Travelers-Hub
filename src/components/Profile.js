import React from 'react';
import { useSelector } from 'react-redux';

const Profile = () => {
  const { missions } = useSelector((state) => state.missions);

  return (
    <div className="profile">
      <div className="missions-profile">
        <h3>My Missions</h3>
        {missions.map((mission) => mission.reserved && <p>{mission.name}</p>)}
      </div>
      <div className="rockets-profile">
        <h3>My Rockets</h3>
        <p>Falcon</p>
      </div>
    </div>
  );
};

export default Profile;
