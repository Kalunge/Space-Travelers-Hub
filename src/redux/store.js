import { createStore, applyMiddleware, combineReducers } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import rockets from './rockets/rockets';

const middleware = [thunk, logger];

const reducer = combineReducers({
  rockets,
});

const store = createStore(reducer, applyMiddleware(...middleware));

export default store;
