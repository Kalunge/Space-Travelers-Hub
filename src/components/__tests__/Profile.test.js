import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../../redux/store';
import Profile from '../Profile';

describe('snapshot test for Profile Component', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Profile />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot('My Rockets');
  });
});
