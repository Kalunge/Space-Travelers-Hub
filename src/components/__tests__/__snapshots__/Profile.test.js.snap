// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test for Profile Component renders correctly: My Rockets 1`] = `
<div
  className="profile"
>
  <div
    className="missions-profile"
  >
    <h3>
      My Missions
    </h3>
  </div>
  <div
    className="rockets-profile ml-5"
  >
    <h3>
      My Rockets
    </h3>
  </div>
</div>
`;
